src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Api.Enduser/Services/ConnectionService.cs

```diff
@@ -0,0 +1,417 @@
+        else
+        {
+            Unreachable();
+        }
+
+        var result = await ConnectionPackageRepository.Get(filter, cancellationToken: cancellationToken);
+        return result.ToList();
+    }
+
+    /// <inheritdoc />
+    public async Task<Result<AssignmentPackage>> AddPackage(Guid from, Guid to, string roleCode, string packageUrn, CancellationToken cancellationToken = default)
+    {
+        var filter = PackageRepository.CreateFilterBuilder()
+            .Add(t => t.Urn, packageUrn, FilterComparer.EndsWith);
+
+        var packages = await PackageRepository.Get(filter, cancellationToken: cancellationToken);
+        var problem = ValidationRules.Validate(ValidationRules.QueryParameters.PackageUrnLookup(packages));
+        if (problem is { })
+        {
+            return problem;
+        }
+
+        var package = packages.First();
+        return await AddPackage(from, to, roleCode, package.Id, "packageUrn", cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public async Task<Result<AssignmentPackage>> AddPackage(Guid from, Guid to, string roleCode, Guid packageId, CancellationToken cancellationToken = default)
+    {
+        return await AddPackage(from, to, roleCode, packageId, "packageId", cancellationToken);
+    }
+
+    private async Task<Result<AssignmentPackage>> AddPackage(Guid from, Guid to, string roleCode, Guid packageId, string queryParamName, CancellationToken cancellationToken)
+    {
+        var dependencies = await GetAssignmentDependencies(from, to, roleCode, cancellationToken);
+        if (ValidateAssignmentData(dependencies.EntityFrom, dependencies.EntityTo, dependencies.Roles) is var problem && problem is { })
+        {
+            return problem;
+        }
+
+        var roleId = dependencies.Roles.First().Id;
+        var filter = AssignmentRepository.CreateFilterBuilder()
+            .Equal(t => t.FromId, from)
+            .Equal(t => t.ToId, to)
+            .Equal(t => t.RoleId, roleId);
+
+        var existingAssignments = await AssignmentRepository.Get(filter, cancellationToken: cancellationToken);
+        problem = ValidationRules.Validate(ValidationRules.QueryParameters.VerifyAssignmentRoleExists(existingAssignments, roleCode));
+        if (problem is { })
+        {
+            return problem;
+        }
+
+        var assignment = existingAssignments.First();
+
+        var connectionPackageFilter = ConnectionPackageRepository.CreateFilterBuilder()
+            .Equal(t => t.FromId, from)
+            .Equal(t => t.ToId, to)
+            .Equal(t => t.PackageId, packageId);
+
+        var connectionPackages = await ConnectionPackageRepository.GetExtended(connectionPackageFilter, cancellationToken: cancellationToken);
+
+        problem = ValidationRules.Validate(
+            ValidationRules.QueryParameters.AnyPackages(connectionPackages, queryParamName),
+            ValidationRules.QueryParameters.PackageIsAssignableByUser(connectionPackages, queryParamName),
```

Comment:
Sliter litt med å forstå logikken som ligger til grunn for å validere om PackageIsAssignableByUser:

Oppslaget av connectionPackages som gjøres her ser ut til å bli gjort basert på From og To assignmenten skal gjøres til (altså om denne eksisterer fra før av?).

Ikke hvor vidt autentisert bruker har rolle-kobling til pakken med CanDelegate=true 🤔

Reply:
validering og logikk jeg tok fra gamle service laget. Tenker du at vi skal fjerne denne sjekken ?